BudgetTracker.BudgetSquirrel.WebApi/Controllers/AuthenticationApiController.cs

```diff
@@ -64,55 +56,46 @@ public async Task<IActionResult> DeleteUser(ApiRequest request)
         /// <returns>
-        /// A JWT Token with claims information for the authenticated user. If
+        /// A Cookie with claims information for the authenticated user. If
         /// the user cannot be authenticated, a 403 will be thrown.
         /// </returns>
         [HttpPost("authenticate")]
-        public async Task<IActionResult> Authenticate(Dictionary<string, string> data)
+        public async Task<IActionResult> Authenticate(Credentials credentials)
         {
             User authenticatedUser;
             UserResponseMessage userResponse;
             try
             {
-                authenticatedUser = await _authenticationService.Authenticate(data["username"], data["password"]);
+                authenticatedUser = await _authenticationService.Authenticate(credentials.Username, credentials.Password);
                 userResponse = new UserResponseMessage(authenticatedUser);
             }
             catch (AuthenticationException e)
             {
-                return new JsonResult(new {
+                return this.BadRequest(new JsonResult(new {
                     error = e.Message
-                });
+                }));
             }
 
-            string securityKey = _authConfig.JWTSecurityKey;
-            SymmetricSecurityKey symSecKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(securityKey));
-            SigningCredentials creds = new SigningCredentials(symSecKey, SecurityAlgorithms.HmacSha256);
+            if (authenticatedUser != null)
+            {
+                var cliamsIdenity = new ClaimsIdentity(authenticatedUser.CreateUserClaims(), CookieAuthenticationDefaults.AuthenticationScheme);
+
+                await HttpContext.SignInAsync(CookieAuthenticationDefaults.AuthenticationScheme,
+                    new ClaimsPrincipal(cliamsIdenity),
+                    new AuthenticationProperties { IsPersistent = true});
+
+                return this.Ok(authenticatedUser);
+            }
 
-            var claims = new List<Claim>();
-            claims.Add(new Claim(ClaimTypes.NameIdentifier, authenticatedUser.Username));
-            
-            DateTime expiration = DateTime.Now.AddHours(_authConfig.JWTDuration);
-            JwtSecurityToken token = new JwtSecurityToken(issuer: _authConfig.JWTIssuer,
-                                            audience: _authConfig.JWTAudience,
-                                            expires: expiration,
-                                            signingCredentials: creds,
-                                            claims: claims);
-            return new JsonResult(new {
-                user = userResponse,
-                token = new JwtSecurityTokenHandler().WriteToken(token),
-                expires = expiration
-            });
+            return this.BadRequest(new JsonResult(new {
+                error = $"Unable to find User with the username {credentials.Username}"
```

Comment:
Didn't you say you didn't want to specify the reason for login failure to prevent account farming?

Reply:
Whooops Not following my own comments lol my bad.